client/src/components/Users/Auth/SignUpCitizen/SignUpCitizen.tsx

```diff
@@ -1,144 +1,97 @@
-    } else {
-      formData.interests.push(interest);
+  const [activeStep, setActiveStep] = useState<number>(0);
+  const [formData, setFormData] = useState(initialFormData);
+  const [submitForm, setSubmitForm] = useState(false);
+  const [user, setUser] = useState(null);
+  const [errorMsg, setErrorMsg] = useState(null);
+
+  useEffect(() => {
+    if (user) {
+      setActiveStep((prevActiveStep) => prevActiveStep + 1);
     }
+  }, [user]);
 
-    setFormData({ ...formData, interests: formData.interests });
-  };
-
-  const makeStateSelectOptions = () => {
-    return US_STATE_NAMES.map((state) => {
-      return <MenuItem value={state}>{state}</MenuItem>;
-    });
-  };
-
-  const handleChange = (evt: React.ChangeEvent<HTMLInputElement>): void => {
-    const { name, value, checked }: { name: string; value: string; checked: boolean } = evt.target;
-    setFormData((fData) => ({
-      ...fData,
-      [name]: name === 'accept_terms' ? checked : value,
-      [name]: name === 'email_notification_opt_out' ? checked : value,
-    }));
-  };
-
-  const handleSelectChange = (event: SelectChangeEvent<string>, child: React.ReactNode): void => {
-    const { name, value }: { name: string; value: string } = event.target;
-    setFormData((fData) => ({
-      ...fData,
-      [name]: value,
-    }));
-  };
-
-  const handleSubmit = async (evt: React.FormEvent) => {
-    evt.preventDefault();
-    console.log(formData);
-    setIsLoading(true);
-    // Backend doesn't need accept_terms. If a user is signed up they have agreed to the terms
-    delete formData.accept_terms;
-    const res = await fetch(`${APP_API_BASE_URL}/auth/register`, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({ ...formData, interests: { names: formData.interests } }),
-    });
-    const data = await res.json();
-    setIsLoading(false);
-    if (data.status === 409) {
-      setEmailError(data.message);
-    } else {
-      setUser(data);
-      handleNext();
+  useEffect(() => {
+    if (submitForm) {
+      registerUserMutation.mutate(formData);
+      setSubmitForm(false);
     }
-  };
+  }, [submitForm]);
+
+  const registerUserMutation = useMutation({
+    mutationFn: Endpoints.userRegister,
+    onSuccess: ({ data: user }) => {
+      setUser(user);
+    },
+    onError: (error: AxiosError) => {
+      setErrorMsg(error?.response?.data?.message);
+    },
+  });
+
+  const handleNext = (newFormData: {}, doSubmit = false) => {
+    setFormData((currFormData) => ({
+      ...currFormData,
+      ...newFormData,
+    }));
 
-  // handleNext and handleBack are also in SignUpUserAndNonProfit, refactor later
-  const handleNext = () => {
-    setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    // eslint-disable-next-line prettier/prettier
+    doSubmit
+      ? setSubmitForm(true)
+      : setActiveStep((prevActiveStep) => prevActiveStep + 1);
```

Comment:
prettier doesn't like this in-line warning disabling comment. See the build errors.

You should be able to just format it in a single line to get rid of the lint warning.

```suggestion
doSubmit ? setSubmitForm(true) : setActiveStep((prevActiveStep) => prevActiveStep + 1);
```